src/ORM/ManyManyThroughList.php

```diff
@@ -25,14 +25,30 @@ class ManyManyThroughList extends RelationList
      * @param string $localKey The key in the join table that maps to the dataClass' PK.
      * @param string $foreignKey The key in the join table that maps to joined class' PK.
      *
+     * @param array $extraFields Ignored for ManyManyThroughList
```

Comment:
This makes me sad :(. I’m guessing it’s because `DataObject::inferReciprocalComponent()` and `DataObject::getManyManyComponents()` don’t know whether they’re constructing `ManyManyList` or `ManyManyThroughList`. Could we add an `if ($manyManyComponent['relationClass'] === ManyManyList::class)` where they’re constructed and pass different args?

Reply:
The arg is ignored because extraFields come from the join model. We aren't ignoring extraFields feature entirely, it's just that the constructor doesn't need it.

> Could we add an if ($manyManyComponent['relationClass'] === ManyManyList::class) where they’re constructed and pass different args?

I'd rather keep it consistent and agnostic to the underlying class constructed. That's the point of soft-coding the relation name.

In the future we may add a third type for PolymorphicManyManyThroughList (e.g.).

Reply:
In 5.x we could switch to named args (one constructor arg, array key in same form produced by DataObjectSchema::manyManyComponent().

Reply:
Okay